#! /bin/sh /usr/share/dpatch/dpatch-run
## 50_dont_downcase_font_names.dpatch by Thomas Weber <thomas.weber.mail@gmail.com>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Don't lowercase the font names; bug fix for #469240
## DP: Upstream patch is 71209cfdaebe, thanks John Eaton

@DPATCH@
diff -urNad octave3.0-3.0.0~/scripts/plot/__go_draw_axes__.m octave3.0-3.0.0/scripts/plot/__go_draw_axes__.m
--- octave3.0-3.0.0~/scripts/plot/__go_draw_axes__.m	2007-12-18 04:48:04.000000000 +0100
+++ octave3.0-3.0.0/scripts/plot/__go_draw_axes__.m	2008-03-04 10:44:04.000000000 +0100
@@ -1602,17 +1602,17 @@
 
 function [f, s, fnt, it, bld] = get_fontname_and_size (t)
   if (isempty (t.fontname))
-    fnt = "helvetica";
+    fnt = "Helvetica";
   else
-    fnt = tolower (t.fontname);
+    fnt = t.fontname;
   endif
   f = fnt;
   it = false;
   bld = false;
-  if (! isempty (t.fontweight) && strcmp (tolower (t.fontweight), "bold"))
+  if (! isempty (t.fontweight) && strcmpi (t.fontweight, "bold"))
     if (! isempty(t.fontangle)
-	&& (strcmp (tolower (t.fontangle), "italic")
-	    || strcmp (tolower (t.fontangle), "oblique")))
+	&& (strcmpi (t.fontangle, "italic")
+	    || strcmpi (t.fontangle, "oblique")))
       f = strcat (f, "-bolditalic");
       it = true;
       bld = true;
@@ -1621,8 +1621,8 @@
       bld = true;
     endif
   elseif (! isempty(t.fontangle)
-	  && (strcmp (tolower (t.fontangle), "italic")
-	      || strcmp (tolower (t.fontangle), "oblique")))
+	  && (strcmpi (t.fontangle, "italic")
+	      || strcmpi (t.fontangle, "oblique")))
     f = strcat (f, "-italic");
     it = true;
   endif
@@ -1640,7 +1640,7 @@
   if (strcmp (fld, "string"))
     [f, s, fnt, it, bld] = get_fontname_and_size (obj);
   else
-    f = "Helvectica";
+    f = "Helvetica";
     s = 10;
     fnt = f;
     it = false;
